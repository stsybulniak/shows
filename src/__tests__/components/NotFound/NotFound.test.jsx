import { fireEvent, render, screen } from '@testing-library/react';
import NotFound from '../../../components/NotFound/NotFound';

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<NotFound />', () => {
  it('NotFound test click with Router trigger', async () => {
    const { baseElement } = render(<NotFound />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockHistoryPush).toHaveBeenCalledWith('/');
    expect(baseElement).toMatchSnapshot();
  });
});
