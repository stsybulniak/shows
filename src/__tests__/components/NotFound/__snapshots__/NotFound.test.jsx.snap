// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> NotFound test click with Router trigger 1`] = `
<body>
  <div>
    <h1>
      Not found
    </h1>
    <button>
      Back home page
    </button>
  </div>
</body>
`;
